jmclient/jmclient/blockchaininterface.py

```diff
@@ -407,33 +408,42 @@ def get_unspent_indices(self, transaction):
         # and return the indices of the others:
         return [i for i, val in enumerate(res) if val]
 
-    def estimate_fee_per_kb(self, N):
+    def estimate_fee_per_kb(self, N, quiet=False):
         """ The argument N may be either a number of blocks target,
         for estimation of feerate by Core, or a number of satoshis
         per kilo-vbyte (see `fee_per_kb_has_been_manually_set` for
         how this is distinguished).
-        In the latter case, it is prevented from falling below the
-        minimum allowed feerate for relay on the Bitcoin network.
+        In both cases it is prevented for falling below the current
+        minimum feerate for tx to be accepted into node's mempool.
         In case of failure to connect, None is returned.
         In case of failure to source a specific minimum fee relay rate
         (which is used to sanity check user's chosen fee rate), 1000
         is used.
         In case of failure to source a feerate estimate for targeted
         number of blocks, a default of 10000 is returned.
         """
+
+        # use the local bitcoin core minimum mempool fee as floor to avoid
+        # relay problems
+        rpc_result = self._rpc('getmempoolinfo', None)
+        if not rpc_result:
+            # in case of connection error:
+            return None
+        mempoolminfee = rpc_result.get('mempoolminfee', 1000)
+        mempoolminfee_in_sat = int(Decimal(1e8) * Decimal(mempoolminfee))
```

Comment:
If you're not 100% sure that the `mempoolminfee` entry will always be returned in the RPC return data, consider adding a `if mempoolminfee` before making this call and falling back otherwise.

Reply:
It is always returned and is never below minimum relay tx fee. https://github.com/bitcoin/bitcoin/blob/d4300a10ddd67ceb06f0a2a27d11b556bc6e4dda/src/rpc/blockchain.cpp#L1546

Reply:
And there is also functional test in Bitcoin Core for this. https://github.com/bitcoin/bitcoin/blob/993e38a4e2fa66093314b988dfbe459f46aa5864/test/functional/mempool_limit.py#L30